src/background/background.ts

```diff
@@ -15,33 +11,54 @@ chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
     if (message.command === "startRecordingBytesTransferred") {
         IBytesRepository.instance.clearBytesTransferred();
 
-        chrome.webRequest.onBeforeRequest.addListener(
-            catchPostRequestBodySize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestBody"]
-        );
-        chrome.webRequest.onBeforeSendHeaders.addListener(
-            catchRequestHeaderSize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestHeaders"]
-        );
-        chrome.webRequest.onCompleted.addListener(
-            catchResponseSize,
-            { urls: ["<all_urls>"], tabId },
-            ["responseHeaders"]
-        );
+        chrome.debugger.attach({ tabId: tabId }, "1.2", function () {
```

Comment:
What does this `debugger.attach` do?

Reply:
Attaches the chrome debugger to a tab, allows the extension to listen to network requests for that tab :)